File: src/main/java/com/backend/app/shared/models/entities/CustomerAddress.java
package com.backend.app.shared.models.entities;

import java.util.Date;


import jakarta.annotation.Nullable;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;
import jakarta.persistence.FetchType;

@Data
@Entity
@Table(name = "customer_addresses")
public class CustomerAddress {

    @Id
    @Column(name = "id")
    private String id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id")
    private Customer customer;

    @Column(name = "address")
    @Nullable
    private String address;

    @Column(name = "latitude")
    @Nullable
    private Double latitude;

    @Column(name = "longitude")
    @Nullable
    private Double longitude;

    @Column(name = "street")
    @Nullable
    private String street;

    @Column(name = "sub_district")
    @Nullable
    private String subDistrict;

    @Column(name = "district")
    @Nullable
    private String district;

    @Column(name = "province")
    @Nullable
    private String province;

    @Column(name = "country")
    @Nullable
    private String country;

    @Column(name = "zipcode")
    @Nullable
    private String zipcode;

    @Column(name = "is_active", columnDefinition = "boolean default true")
    private Boolean isActive;

    @Column(name = "created_date")
    private Date createdDate;

    @Column(name = "created_by")
    private String createdBy;

    @Column(name = "updated_date")
    private Date updatedDate;

    @Column(name = "updated_by")
    private String updatedBy;
}

File: user-svc/UserService/Models/Entities/CustomerAddress.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Reflection;

namespace UserService.Models.Entities
{
    [Table("customer_addresses")]
    public class CustomerAddress
    {
        [Key]
        [Column("id")]
        [Required]
        public required string Id { get; set; }

        [Column("customer_id")]
        [Required]
        public required string CustomerId { get; set; }

        public Customer Customer { get; set; }

        [Column("address")]
        [Required]
        public required string Address { get; set; }

        [Column("latitude")]
        [Required]
        public double Latitude { get; set; }

        [Column("longitude")]
        [Required]
        public double Longitude { get; set; }

        [Column("street")]
        public string? Street { get; set; }

        [Column("sub_district")]
        public string? SubDistrict { get; set; }

        [Column("district")]
        public string? District { get; set; }

        [Column("province")]
        public string? Province { get; set; }

        [Column("country")]
        public string? Country { get; set; }

        [Column("zipcode")]
        public string? Zipcode { get; set; }

        [Column("is_active")]
        [Required]
        public bool IsActive { get; set; } = true;

        [Column("created_date")]
        [Required]
        public DateTime CreatedDate { get; set; }

        [Column("created_by")]
        [Required]
        public required string CreatedBy { get; set; }

        [Column("updated_date")]
        [Required]
        public DateTime UpdatedDate { get; set; }

        [Column("updated_by")]
        [Required]
        public required string UpdatedBy { get; set; }

        public override string ToString()
        {
            var properties = GetType().GetProperties(BindingFlags.Public | BindingFlags.Instance);
            var values = properties.Select(p => $"{p.Name}: {p.GetValue(this) ?? "null"}");
            return $"User: {{ {string.Join(", ", values)} }}";
        }
    }

}
